### PR TITLE
Add MSALErrorConverter to convert MSIDError

### PR DESCRIPTION
**Pull request by jasoncoolmax**
ADAL and MSAL use different error domains and error codes. A translation between MSIDError and ADAL/MSAL error object is needed.

### CODE REVIEW

File: MSAL/src/MSALErrorConverter.m [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+@implementation MSALErrorConverter
+
++ (void)initialize
+{
+    s_errorDomainMapping = @{
+                             MSIDErrorDomain : MSALErrorDomain
+                             };
+    
+    s_errorCodeMapping = @{
+                           //sample format is like @"MSIDErrorDomain|-10000":@"MSALErrorDomain:-20000"
```
Comment (oldalton) [inline MSAL/src/MSALErrorConverter.m]: I'd prefer using the same delimiter, e.g. | in both key and value

Comment (oldalton) [reply MSAL/src/MSALErrorConverter.m]: Another question is why are both domain and code necessary in the value part? I agree that it's necessary for the key, but for domain mapping there's already s_errorDomainMapping. Another alternative is to remove s_errorDomainMapping and get domain from this mapping.

Comment (jasoncoolmax) [reply MSAL/src/MSALErrorConverter.m]: Agree.

Regarding the alternative, current design can work for the scenario where we only want to map the domain but keep the error code, like NSOSStatusErrorDomain->ADKeychainErrorDomain. For such a scenario, we only need to fill up s_errorDomainMapping but no need to fill up  s_errorCodeMapping, what do you think?

Comment (oldalton) [reply MSAL/src/MSALErrorConverter.m]: I agree, it makes sense to keep s_errorDomainMapping. Just let's remove the domain from the value for s_errorCodeMapping unless there's an actual use for it. Thanks :)

**Pull request merged**